src/components/Contractor/Payments/CreatePayment/CreatePayment.tsx

```diff
@@ -15,41 +28,155 @@ export function CreatePayment(props: CreatePaymentProps) {
   )
 }
 
-export const Root = ({ companyId, dictionary, onEvent, children }: CreatePaymentProps) => {
+export const Root = ({ companyId, dictionary, onEvent }: CreatePaymentProps) => {
   useComponentDictionary('Contractor.Payments.CreatePayment', dictionary)
+  const { Modal } = useComponentContext()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const [paymentDate, setPaymentDate] = useState<string>(
+    new Date().toISOString().split('T')[0] || '',
+  )
+
+  const { mutateAsync: createContractorPaymentGroup } = useContractorPaymentGroupsCreateMutation()
+
+  const { data: contractorList } = useContractorsListSuspense({ companyUuid: companyId })
+  const contractors = (contractorList.contractorList || []).filter(
+    contractor =>
+      contractor.isActive &&
+      contractor.onboardingStatus === ContractorOnboardingStatus.ONBOARDING_COMPLETED,
+  )
+  const initialContractorPayments: ContractorPayments[] = useMemo(
+    () =>
+      contractors.map(contractor => ({
+        contractorUuid: contractor.uuid,
+        paymentMethod: contractor.paymentMethod || 'Direct Deposit',
+        wage: 0,
+        hours: 0,
+        bonus: 0,
+        reimbursement: 0,
+      })),
+    [contractors],
+  )
+  const [virtualContractorPayments, setVirtualContractorPayments] =
+    useState<ContractorPayments[]>(initialContractorPayments)
+  //TODO: fix totals - they are not correct
+  const totals = useMemo(
+    () =>
+      virtualContractorPayments.reduce<{
+        wage: number
+        bonus: number
+        reimbursement: number
+        total: number
+      }>(
+        (acc, contractor) => {
+          return {
+            wage: acc.wage + contractor.wage!,
+            bonus: acc.bonus + contractor.bonus!,
+            reimbursement: acc.reimbursement + contractor.reimbursement!,
+            total: acc.total + contractor.wage! + contractor.bonus! + contractor.reimbursement!,
+          }
+        },
+        { wage: 0, bonus: 0, reimbursement: 0, total: 0 },
+      ),
+    [virtualContractorPayments],
+  )
 
-  const onSaveAndContinue = () => {
-    onEvent(componentEvents.CONTRACTOR_PAYMENT_REVIEW)
+  const formMethods = useForm<EditContractorPaymentFormValues>({
+    resolver: zodResolver(EditContractorPaymentFormSchema),
+    defaultValues: {
+      wageType: 'Hourly',
+      hours: 0,
+      wage: 0,
+      bonus: 0,
+      reimbursement: 0,
+      paymentMethod: 'Direct Deposit',
+      hourlyRate: 0,
+      contractorUuid: '',
+    },
+  })
+
+  const onSaveAndContinue = async () => {
+    const response = await createContractorPaymentGroup({
+      request: {
+        companyId,
+        requestBody: {
+          checkDate: new RFCDate(paymentDate),
+          contractorPayments: virtualContractorPayments,
+          creationToken: crypto.randomUUID(),
```

Comment:
is this required? would it be preferable for it to get assigned on the backend if possible?